var/spack/repos/builtin/packages/meme/package.py

```diff
@@ -35,9 +35,26 @@ class Meme(AutotoolsPackage):
 
     version('4.11.4', '371f513f82fa0888205748e333003897')
 
+    variant('mpi', default=False, description='Enable MPI support')
+    variant('serial', default=False, description='Disable MPI support')
+
     depends_on('zlib', type=('link'))
     depends_on('libxml2', type=('link'))
     depends_on('libxslt', type=('link'))
     depends_on('libgcrypt', type=('link'))
     depends_on('perl', type=('build', 'run'))
     depends_on('python@2.7:', type=('build', 'run'))
+    # openmpi support
+    depends_on('openmpi@2:', when='+mpi', type=('link'))
```

Comment:
I have this:

```
    # mpi support
    depends_on('mpi@2:', when='+mpi', type=('link'))
    depends_on('mpi@3:', when='+mpich', type=('link'))
```

Though:

```
Input spec
--------------------------------
meme+mpich

Normalized
--------------------------------
meme+mpich
    ^libgcrypt
        ^libgpg-error
    ^libxml2
        ^pkg-config@0.9.0:
        ^xz
        ^zlib
    ^libxslt
    ^mpi@3:
    ^perl
        ^gdbm
            ^readline
                ^ncurses
    ^python@2.7:
        ^bzip2
        ^openssl
        ^sqlite

Concretized
--------------------------------
meme@4.11.4%gcc@4.4.6~mpi+mpich~serial arch=linux-rhel6-x86_64 
    ^libgcrypt@1.6.2%gcc@4.4.6 arch=linux-rhel6-x86_64 
        ^libgpg-error@1.21%gcc@4.4.6 arch=linux-rhel6-x86_64 
    ^libxml2@2.9.4%gcc@4.4.6~python arch=linux-rhel6-x86_64 
        ^pkg-config@0.29.2%gcc@4.4.6+internal_glib arch=linux-rhel6-x86_64 
        ^xz@5.2.3%gcc@4.4.6 arch=linux-rhel6-x86_64 
        ^zlib@1.2.11%gcc@4.4.6+pic+shared arch=linux-rhel6-x86_64 
    ^libxslt@1.1.29%gcc@4.4.6 arch=linux-rhel6-x86_64 
    ^openmpi@2.1.1%gcc@4.4.6~cuda fabrics=verbs ~java schedulers= ~sqlite3~thread_multiple+vt arch=linux-rhel6-x86_64 
        ^hwloc@1.11.6%gcc@4.4.6~cuda+libxml2+pci arch=linux-rhel6-x86_64 
            ^libpciaccess@0.13.4%gcc@4.4.6 arch=linux-rhel6-x86_64 
                ^libtool@2.4.6%gcc@4.4.6 arch=linux-rhel6-x86_64 
                    ^m4@1.4.18%gcc@4.4.6+sigsegv arch=linux-rhel6-x86_64 
                        ^libsigsegv@2.11%gcc@4.4.6 arch=linux-rhel6-x86_64 
                ^util-macros@1.19.1%gcc@4.4.6 arch=linux-rhel6-x86_64 
    ^perl@5.24.1%gcc@4.4.6+cpanm arch=linux-rhel6-x86_64 
        ^gdbm@1.13%gcc@4.4.6 arch=linux-rhel6-x86_64 
            ^readline@7.0%gcc@4.4.6 arch=linux-rhel6-x86_64 
                ^ncurses@6.0%gcc@4.4.6~symlinks arch=linux-rhel6-x86_64 
    ^python@2.7.13%gcc@4.4.6+shared~tk~ucs4 arch=linux-rhel6-x86_64 
        ^bzip2@1.0.6%gcc@4.4.6+shared arch=linux-rhel6-x86_64 
        ^openssl@1.0.2k%gcc@4.4.6 arch=linux-rhel6-x86_64 
        ^sqlite@3.18.0%gcc@4.4.6 arch=linux-rhel6-x86_64 
```

OpenMPI does not provide mpi@3 though MPICH does so why do I see " ^openmpi@2.1.1%gcc@4.4.6~cuda fabrics=verbs ~java schedulers="